packages/controller/src/main.ts

```diff
@@ -3032,18 +3032,43 @@ async function processMessage(msg: ioBroker.SendableMessage): Promise<null | voi
                         if (msg.message.link) {
                             if (!error && base64) {
                                 const buff = Buffer.from(base64, 'base64');
-                                states!.setBinaryState(`${hostObjectPrefix}.zip.${msg.message.link}`, buff, err => {
-                                    if (err) {
-                                        sendTo(msg.from, msg.command, { error: err }, msg.callback);
-                                    } else {
-                                        sendTo(
-                                            msg.from,
-                                            msg.command,
-                                            `${hostObjectPrefix}.zip.${msg.message.link}`,
-                                            msg.callback
-                                        );
-                                    }
-                                });
+                                if (msg.message.storeToFile) {
+                                    objects!.writeFile(
+                                        msg.message.storeToFile,
+                                        `zip/${msg.message.link}`,
+                                        buff,
+                                        err => {
+                                            if (err) {
+                                                sendTo(msg.from, msg.command, { error: err }, msg.callback);
+                                            } else {
+                                                sendTo(
+                                                    msg.from,
+                                                    msg.command,
+                                                    `${msg.message.storeToFile}/zip/${msg.message.link}`,
+                                                    msg.callback
+                                                );
+                                            }
+                                        }
+                                    );
+                                } else {
+                                    logger.warn(
+                                        `${hostLogPrefix} Saving in binary states is deprecated.` +
+                                            'Please add the "storeToFile" attribute to request (with e.g. "admin.0" value) to save ZIP in file'
```

Comment:
we should add the "location" also here because else noone will have a chance to find ourt who sent what to fix ...
so ${hostObjectPrefix}.zip.${msg.message.link} should be added to log message too

Reply:
Added